### PR TITLE
Rust add kerberos 04

### PR DESCRIPTION
**Pull request by chifflier**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Describe changes:
- Update of #3345, #3371 and #3380  (rebase + rewrite)
- Fix typos in doc, and add example for events
- Make suricata.yaml logging option depend on Rust, and move it near the other Rust loggers
- Add TCP buffering code, and properly handle fragmented messages by parsing Record Mark field and buffering messages. If the message is not fragmented, no copy is done.

### CODE REVIEW

File: suricata.yaml.in [hunk 1]
```diff
@@ -808,6 +810,9 @@ pcap-file:
 # "detection-only" enables protocol detection only (parser disabled).
 app-layer:
   protocols:
+    krb5:
+      enabled: @rust_config_enabled@
+      warn-weak-crypto: yes
```
Comment (jasonish) [inline suricata.yaml.in]: Is this an app-layer option instead of just disabling the rule for performance reasons?

**Comment by jasonish** [comment]
This is looking good. 2 points of discussion though:

- I think I'd rather see simply "weak" rather than "weak_crypto" in the output. Or perhaps "weak_encryption".

- Is there any point is using event_type of "krb" and setting a version field to 5? Or is Kerberos 5 really its own thing?

Thanks.

File: suricata.yaml.in (hunk 1 above)
Comment (chifflier) [reply suricata.yaml.in]: Hmm, I think I missed the easiest solution. I'll remove this option, disabling the rule should be enough

**Comment by chifflier** [comment]
Hi Jason,
* I'd prefer "weak_encryption". In some other protocols there are other cryptographic parameters that are tested (hash algorithm, signature algorithm, etc), it may be interesting to know what is weak
* The protocol name and version number is a good one. Kerberos 5 was released in the 80s and never had a version update, so I tend to consider it as its own thing
What would you prefer ?

**Comment by jasonish** [comment]
> I'd prefer "weak_encryption". In some other protocols there are other cryptographic parameters that are tested (hash algorithm, signature algorithm, etc), it may be interesting to know what is weak

Ok, then I think weak_encryption is good. It makes it even more clear as to whats weak. I think "crypto" is often used as a generic term.

> The protocol name and version number is a good one. Kerberos 5 was released in the 80s and never had a version update, so I tend to consider it as its own thing
> What would you prefer ?

krb5 seems appropriate then. So leave the name as is. Thanks.

**Comment by chifflier** [comment]
Replaced by #3391

**Pull request closed without merging**